### DISCUSSION TITLE
[Q&A] database inventory

### DISCUSSION THREAD

**Discussion by b-keller** [glpi-project/glpi-agent#727] [Q&A] [answered]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

Hi,

I would like to know more about the possible parameters for the database inventory.

I found this:

https://glpi-agent.readthedocs.io/en/latest/man/glpi-agent.html#inventory-task-specific-options
```
--credentials=CREDENTIALS
Setup credentials for database inventory
CREDENTIALS should be a list of "key:value" separated by commas like in:
For example: --credentials="type:login_password,login:root,password:********,use:postgresql,params_id:0"
```

- are there several login "types"?
- how to use the "params_id" parameter?
- how to keep the login parameters in the agent's conf file?
- is it possible to exclude certain base from the inventory?

I'm having a bit of trouble making an inventory of an oracle base on windows,
the identification string is of type 
`user/password@service`
how to pass this ?

thx

**Comment by g-bougard** [comment]
Hi @b-keller 

sorry, this option is not well documented and actually you'll have to check the code to know what parameters can be used for each kind of database and how they are used.

To answer your questions:
 * yes, there are different login types as this depend on database kind and possible login options. If people wants another type, this still can be developed.
 * "params_id" is an identifier in the case you provide more than one credentials, it is essentially used to identified in log which credentials is in use. Of course, you must not use the same params_id when providing more than one credentials to the agent.
 * it is not possible to keep credentials in conf.
 * from the agent perspective, it just runs an inventory and will report every database it detects. If you don't want to see some databases, maybe you can use credentials which only authorize to see the wanted databases. Or maybe you can put them in inventory blacklist on glpi side.

You may be interested by this plugin which permits to manage databases credentials server-side: https://plugins.glpi-project.org/#/plugin/databaseinventory
With this plugin, you don't have to handle the params_id value as it is automatically set.

Maybe this plugin can be the right solution for your Oracle case on windows. Otherwise, you may have to escape some charaters on the commandline option.
For Oracle, credentials configuration is analyzed here: https://github.com/glpi-project/glpi-agent/blob/de5f5b515e2352c3fa864f5cbe9cd8f735ae6fff/lib/GLPI/Agent/Task/Inventory/Generic/Databases/Oracle.pm#L440
and the credentials will be used with sqlplus command.

**Comment by b-keller** [comment, marked as answer]
@g-bougard 
ok, so I searched and found the missing parameter

`--credentials=use:oracle,type:login_password,login:myuser,password:mypassword,host:servicename`

it works

I'll try Database inventory plugin to manage credentials in glpi

thx
